pages/c++/std/partition.md

```diff
@@ -0,0 +1,10 @@
+> Sort array elements before and after a condition
+> More information: <https://en.cppreference.com/w/cpp/algorithm/partition>
+
+- Partition an array:
+
+```
+std::partition(array.begin(), array.end(), [](const {{int}}) &a) {
+  return {{condition}}
```

Comment:
Needs ideas for placeholder syntax that doesn't overlap with any programming syntax.